libraries/joomla/factory.php

```diff
@@ -326,9 +301,9 @@ public static function getDbo()
 	{
 		if (!self::$database)
 		{
-			self::$database = self::createDbo();
+			// TODO where should the database exactly being attached to?
+			self::$database = self::$application->getContainer()->get('JDatabaseDriver');
```

Comment:
For the moment I'd say leave it here and as things get a bit more defined we can sort out what to do with this.